myapp/app/controllers/application_controller.rb

```diff
@@ -28,5 +29,16 @@ def require_login
       redirect_to login_path
     end
   end
+
+  def redirect_to_maintenance
+    if File.exist?(Rails.root.join('tmp', 'maintenance.txt'))
+      return if request.path == maintenance_path
+      redirect_to maintenance_path
+    end
+  end
+
+  def maintenance
```

Comment:
Since this maintenance method is marked as private, it doesn't seem to be recognized as a controller action. You can confirm that it continues to work even if you delete this method or raise an exception within it.